### ISSUE TITLE
Option to output outdated games

### ISSUE THREAD

**Issue by Elionna** [julesontheroad/NSC_BUILDER#114]
It would be cool if this app could check my NSP/XCI library and output which games need to be updated (similar to how doge functions).

The version info can be checked from tinfoil.media sources:

update list:
https://tinfoil.media/repo/db/versions.txt

titledb:
https://tinfoil.media/repo/db/titles.json

This app already covers mostly everything one would require regarding a switch game library and this is the only feature left that will make it the only tool I need. Thanks.

**Comment by julesontheroad** [comment]
This is coming soon, i was waiting to add libraries to the interface before adding that feature.

**Comment by jasii** [comment]
Is this currently in the the latest version .99? I would like to have NSCB monitor my local library on my windows desktop and check against a drive library and if the drive has a newer update it'd auto download.

**Comment by julesontheroad** [comment]
Not for local yet but as you probably guessed that's the plan.  I'll add the autoupdater i added on mtp for local, the checker against the database and the idea is to add a verification database. For local i was thinking about this workflow:
1.- Renamer if games don't have titleid. This one will be based on the database dictionary to add more custom tags like buildversion, masterkey, keygeneration, etc... as the user liking but i think the ideal is to keep at least [titleid][version] for normal files and the content tags for multis (nG+nU+nD)
2.- Verification database, to keep track of the files in a database.
3.- Updater, check if there are new files on other remote, local and google drive. Local could seem useless but you can mount other providers like onedrive or ftp with Raidrive or rclone or may want to clone between HDDs so I think that is useful too.
4.- Option to keep old updates and dlcs in another folder and to keep things that don't pass verification in other folder too with the  log so the user can decide.

Since it's easier to add stuff at the beginning let me know what you think about that workflow.

**Comment by jasii** [comment]
That sounds great because I like keeping my files local, but only my favorite ones.

I would throw in a blacklist based on titleid, to prevent dling language pack dlcs and other unwanted files.

It'd be cool to have
1. Wanted list for a base titleid, Downloads the base 
2. Once base is in local, grab available updates and dlc from remote drive.

Thanks for making an awesome program. Once this autoupdater for local is added it's totally going to replace my old doge / nut setup. I appreciate all your hard work.

**Comment by Masamune3210** [comment]
I'm currently creating multi's for the entire switch library as a project for my archive, so the whole 'place files in another folder if they fail verification' sounds amazing! The checking for updates vs the known latest of files sounds good too, as I am currently using a different program to check for missing/out of date files and sorting the base plus all the dlc into separate folders by base

**Comment by Masamune3210** [comment]
I'll make a new issue if needed to track it, but could an option be added to sort all dlc and updates of a game into a folder along with its base if found in a folder?

**Comment by julesontheroad** [comment]
@Masamune3210 that was proposed before i actually have it on mind. It was proposed to rename the folders using the content tags too. Like (nG+nU+nD) i postponed it since i wasn't sure if having a folder with changing names was a good idea in case i linked verification by path. Of course i can derive the path by removing the tags but i'm on the edge about that approach.
About games per folder legacy had something similar or maybe it was when it was nut_bc and only removed tickets, i don't remember now.

**Comment by julesontheroad** [comment]
@jasii Actually squirrel has an undocumented blacklist, whilelist and a system to compare lists and generate another based on missing updates and dlcs. It kinda have some functions to work with text files and automate things in batches just didn't implemented it in NSCB since is the friendly approach.

**Comment by Masamune3210** [comment]
Changing dir names would probably end poorly as most people would probably get confused or have issues if they are doing things with scripts or other programs. The way nsp-update does it is it creates a folder with the base games name if found in the folder, and then adds the base game plus one copy of any dlc or updates in the folder. It's useful for if you have a folder of nsp files, you can run it and the remaining files that are outside the created folders are either duplicates or missing their base, if that makes any sense

**Comment by julesontheroad** [comment]
Yeah you're right about external scripts maybe. In my case it wouldn't really matter since my list function is recursive, anyway i may still consider it giving it as an option, we'll see.

**Comment by julesontheroad** [comment]
@mspykerez not sure as part of this i also want to reform part of the old batches and squirrel. I may do it by steps though so i don't have to retest everything so i may start with the checking stuff and then add the rest while i reform functions. Anyway i'll be busy this weekend so I'll probably start next week.

**Comment by sepej** [comment]
Is this possible to do now with 1.01? If so, how?

**Comment by jasii** [comment]
Are you able to do this now with the 1.01 release?

**Comment by julesontheroad** [comment]
No, 1.01 is more like a reaction to some reported bugs, most of the new stuff was actually added a while back to the beta. I want that to be included once the rework branch is finished since it should make updates easier.

**Comment by mspykerez** [comment]
It's been over a year since this was requested...

Just to clear something up, the plan is to implement a way to import your local library of games, including updates and DLC and let you browser over them and check things like missing updates and DLC, correct ?

I was wondering if it is possible to also provide a full list of the nintendo switch games that we are missing? That is games that have been released or have been just released or even upcoming games. I would very much like to be able to browser said list and check for games that I'd like to try and even better the ability to find similar games to the ones we have.

ps. Is there any reference to the eshop page of the game we load on the GUI? I've looked and didn't find anything.
